### ISSUE TITLE
📎 Create Homebrew formula

### ISSUE THREAD

**Issue by nhedger** [biomejs/biome#209]
### Description

This task tracks the publication of Biome to Homebrew.

- [x] Create Homebrew formula
- [x] Submit to homebrew/core
- [x] Update docs
- [ ] Update workflow to release a new version automatically

Discussed in https://github.com/biomejs/biome/discussions/197

- Waiting on #211 to be merged

**Comment by nhedger** [comment]
Please assign me to the issue.

**Comment by nhedger** [comment]
I've got a working formula, but we should wait for the next release of the CLI (which renames `rome_cli` to `biome_cli`) before submitting it, otherwise, we'll have to update the formula with the new `biome_cli` name afterwards.

```rb
class Biome < Formula
  desc "Toolchain of the web"
  homepage "https://biomejs.dev/"
  url "https://github.com/biomejs/biome.git",
      tag:      "cli/v1.1.2",
      revision: "cde65810dc5a5fa0be30da9ddfe27d2769e71731"
  license "MIT"

  depends_on "rust" => :build

  def install
    cd "crates/rome_cli" do
      system "cargo", "install", "--bin", "biome", *std_cargo_args
    end
  end

  test do
    unformatted = "const x = 1"

    # Write the unformatted code to a file
    (testpath/"test.js").write(unformatted)

    # Run the formatter
    system "#{bin}/biome", "format", "--semicolons=always", "--write", (testpath/"test.js")

    # Read the formatted file
    formatted = (testpath/"test.js").read

    expected = "const x = 1;"

    # Assert that the expected and formatted match
    assert_match expected, formatted
  end
end

```

### Todo

Once a new release is made for the CLI:

- [ ] Rename the tag
- [ ] Update the revision
- [ ] Rename `rome_cli` to `biome_cli`

**Comment by ematipico** [comment]
How will the publishing process work with homebrew?

At the moment we compile for production using GitHub actions, and they contain environment variables required during the release (version, release flags, etc.)

**Comment by nhedger** [comment]
This can either be done manually after each release by running the following command:

```bash
brew bump-formula-pr \
  --strict biome \
  --tag="cli/vX.Y.Z" \
  --revision="<commit sha>"
```

Or can be automated in GitHub Actions using the following GitHub Action, that wraps the previous script.

https://github.com/marketplace/actions/homebrew-bump-formula

**Comment by nhedger** [comment]
We might be able to add a `livecheck` block to the formula so that Homebrew's bot automatically maintains this for us. I'll explore this solution and report my findings here.

**Comment by nhedger** [comment]
The following livecheck block will detect new versions of the CLI.

- Ensures that only releases of the CLI are detected
- Excludes pre-releases and drafts

```rb
  livecheck do
    url :stable
    regex(/^cli\/v(\d+(?:\.\d+)+)$/i)
    strategy :github_releases do |json, regex|
      json.map do |release|
        next if release["draft"] || release["prerelease"]

        match = release["tag_name"]&.match(regex)
        next if match.blank?
        
        match[1]
      end
    end
  end
```

**Comment by nhedger** [comment]
v1.2.0 of the CLI has been released, so I'll proceed with this.

**Comment by nhedger** [comment]
Biome CLI v1.2.2 is now available on Homebrew.

https://formulae.brew.sh/formula/biome

I'll keep this issue open for now because we still need to integrate the formula version bumping strategy in the release workflow.

**Comment by ematipico** [comment]
Can we close this @nhedger ? What's missing?

**Comment by nhedger** [comment]
I still have to update the workflow to automatically bump the version on homebrew's side when we release a new version.

**Comment by nhedger** [comment]
Closing in favor of https://github.com/biomejs/biome/issues/568
